### PR TITLE
CISCO: Enhance sonic-config-engine to support L1 & L3 configurations

### PR DESCRIPTION
**Pull request by VenkatCisco**
#### Why I did it
This provides support for: https://github.com/Azure/sonic-buildimage/issues/7074. 

#### How I did it
Extend sonic-config-engine/config_samples.py to provide support for l1 & l3 

#### How to verify it
Verify system bring up with L1 & L3 configuration 

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [x] 201911
- [x] 202006
- [x] 202012

### CODE REVIEW

File: src/sonic-config-engine/config_samples.py [hunk 1]
```diff
@@ -95,7 +128,9 @@ def generate_l2_config(data):
 _sample_generators = {
         't1': generate_t1_sample_config,
         'l2': generate_l2_config,
-        'empty': generate_empty_config
+        'empty': generate_empty_config,
+        'l1': generate_l1_config,
+        'l3': generate_l3_config
```
Comment (lguohan) [inline src/sonic-config-engine/config_samples.py:133]: can you explain what is l1, l3 config?

File: src/sonic-config-engine/config_samples.py [hunk 2]
```diff
@@ -9,6 +9,39 @@
...
+        data['PORT'][port]['admin_status'] = 'up'
+        data['PORT'][port]['mtu'] = '9100'
+    return data;
+
+def generate_l3_config(data):
+    data['DEVICE_METADATA']['localhost']['hostname'] = 'sonic'
+    data['DEVICE_METADATA']['localhost']['type'] = 'LeafRouter'
+    data['DEVICE_METADATA']['localhost']['bgp_asn'] = '65100'
+    data['LOOPBACK_INTERFACE'] = {"Loopback0": {},
+                                  "Loopback0|10.1.0.1/32": {}}
```
Comment (lguohan) [inline src/sonic-config-engine/config_samples.py:35]: can you remove those hardcoded 'type', 'bgp_asn' and 'LOOPBACK_INTERFACE'?

Comment (VenkatCisco) [reply src/sonic-config-engine/config_samples.py:35]: Please review https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-config-engine/config_samples.py 
The new methods for L1 & L3 are just extensions to the existing methods and has the minimum set of parameters required for L1 & L3 System bringup.

Comment (VenkatCisco) [reply src/sonic-config-engine/config_samples.py:35]: Tracked thru #7637

**Pull request closed without merging**